### PR TITLE
Netvfy agent cleanup

### PR DESCRIPTION
**Pull request by si74**
Note this PR has three key components: 
1. adding a makefile to ensure that we are gofmting, running go vet, static checking, etc - this is catching a lot of errors (I noticed this due to my vs code auto-linter extensions) 
2. adding more instructions to the Makefile
3. making some minor agent/arp package fixes based on the Makefile
4. go vendoring all the dependencies

### CODE REVIEW

File: arp.go [hunk 1]
```diff
@@ -62,7 +61,7 @@ func (t *ArpTable) Add(IP string) error {
 
 	// Check if an entry already exist before adding one.
 	_, ok := t.ArpMap.Load(IP)
-	if ok == false {
```
Comment (si74) [inline arp.go:65]: linter fail plus non idiomatic go

File: arp.go [hunk 2]
```diff
@@ -114,8 +113,6 @@ type ARPQueue struct {
 	list.List
 	// max length of buffer
 	length uint
-	// logger
-	ll log.Logger
```
Comment (si74) [inline arp.go:118]: we seemed to not actually be using the logger

File: arp.go (hunk 1 above)
Comment (nicboul) [reply arp.go:65]: you mean staticcheck ? can you show me what it says 🤔 
is there a reference what is idiomatic go ?

**Comment by si74** [comment]
Yup running staticcheck ./... returned an error message for cases of
checking var == false. I would suggest downloading the tool to run the make
commands locally (https://staticcheck.io/). On a personal note, this is
also a pattern I almost never see in all the Go code I've written.
Typically one just does a !varName for booleans.

File: arp.go (hunk 1 above)
Comment (si74) [reply arp.go:65]: I would suggest installing -> https://staticcheck.io/

And yup! sorry the golinter was replaced with staticcheck a few years ago. 

Basically shows an error of this form: 
Screen Shot 2021-06-06 at 12 23 51 AM

Unlike C, I don't know if there is a definitive style guide per se but the staticcheck does often catch of lot of these non idiomatic things. As well as gofmt. A lot of what I have learned re: idiomatic Go is from either of those, just the Go language spec, or from writing work with contributors.

**Comment by si74** [comment]
@nicboul, Idecided to do a bit more research into idiomatic Go documentation and there is this -> https://golang.org/doc/effective_go#blank

I noticed, however, that this doesn't contain a things that I've come to write/use very frequently over the last few years: 
1. struct type assertions for interfaces -> `var _  TestInterface = (*MyType)(nil)`
2. declaring a slices capacity up front whereas it still has the length 0 ->  `val := make([]int, 0, 5)` is preferable when possible (otherwise the underlying array needs to be resized as the slice is appended to) 
3. Typically use `ok` as a boolean variable name when checking to see if a map contains a key. `val, ok := mapA["keyA"]` 

Some of these are more along the line of style whereas others are optimizations (that the compiler would not otherwise do) and ways to force compile-time checks vs. run time checks. 

So I'd say that the best approach might be to: 
1 - Check what the effective Go style guide says
2 - Rely on govet, staticcheck, and gofmt
3 - Consider other idiomatic Goisms from experience

**Comment by si74** [comment]
@nicboul fell down a rabbithole but I did actually find some more links re: golang style (super helpful for me to take a look at too in the future - curious how much of this is already integrated into staticcheck): 

1. Uber style guide: https://github.com/uber-go/guide/blob/master/style.md (Taking a quick look at it, it is quite similar to patterns I'm seeing at Fastly.) 
2. Google style guide: https://github.com/golang/go/wiki/CodeReviewComments (Has some really good stuff too - also it's from the inventors of Go so can't be wrong really)

**Pull request merged**